config/routes.rb

```diff
@@ -1,3 +1,6 @@
 Rails.application.routes.draw do
+  root 'xml_parsers#index'
+
+  post '/upload-xml' => 'xml_parsers#upload'
```

Comment:
Since we're not making use of RoR's internal CRUD functionality nor tying the controller name to a concept that's an entity, I would suggest renaming this controller into something else, e.g. `parser_controller.rb` or such.